packages/catalog-model/src/kinds/relations.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+Naming rules for relations in priority order:
+
+1. Use at most two words. One main verb and a specifier, e.g. "ownerOf"
+2. Reading out "<source-kind> <type> <target-kind>" should make sense in English.
+3. Maintain symmetry between pairs, e.g. "ownedBy" and "ownerOf" rather than "owns".
+*/
+
+/**
+ * An ownership relation where the owner is usually an organizational
+ * entity (user or group), and the other entity can be anything.
+ */
+export const RELATION_OWNED_BY = 'ownedBy';
+export const RELATION_OWNER_OF = 'ownerOf';
+
+/**
+ * A relation with an API entity, typically from a component or system
+ */
+export const RELATION_IMPLEMENTED_BY = 'implementedBy';
+export const RELATION_IMPLEMENTS = 'implements';
+
+/**
+ * A relation denoting a dependency on another entity.
+ */
+export const RELATION_DEPENDS_ON = 'dependsOn';
+export const RELATION_DEPENDENCY_OF = 'dependencyOf';
+
+/**
+ * A parent/child relation to build up a tree, used for example to describe
+ * the organizational structure between groups.
+ */
+export const RELATION_PARENT_OF = 'parentOf';
+export const RELATION_CHILD_OF = 'childOf';
```

Comment:
I wonder if relationships always have a fixed set of allowed source-kind and  target-kind, or can I use them in any combination?

Like TEAM-A parentOf COMPONENT-B? 

For systems/domains, would I use the parentOf relationship, or create a new set of terms like partOf/contains?

Reply:
Current thinking is that they are not tied to the source/target kinds. Agree that `parentOf`/`childOf` could potentially be used for domains/systems. Not sure yet whether we actually want to strive for fewer relations though, as in we might have both parent/child and part/contains if they each make the most sense for each usage.

Reply:
Hm tricky. We could totally use component.childOf.system and system.parentOf.component if we wanted to keep the number of relations small, but I'm wondering if it really conceptually feels like a parent/child relation. Then what about transitive properties - component.childOf.domain? 

I think I'd prefer partOf and contains, instead (both direct and transitive) for component<->system<->domain

Reply:
My original idea for relation names was to make them more descriptive to the point of basically describing the target - component.exposesApi.api, component.belongsToSystem.system, etc

Reply:
One more thought about transitivity. When it comes to the old Group fields ancestors and descendants, I wonder if it's worth expressing those as relations - as well as users' direct and indirect memberships.

freben.memberOf.team-name could be the direct one, freben.directOrIndirectMemberOf.team-name as well as freben.directOrIndirectMemberOf.business-unit the other one? or would we hijack partOf for that?

team-name.childOf.business-unit, team-name.partOf.org-root?

One problem here could be relation explosions; org-root.hasDirectOrIndirectMember.* is every single user in the company.

Reply:
An option is to allow the same kind of query to be constructed, as in querying for direct and transitive relations with an infinite or fixed depth, duplicates removed ofc